analysis_runner/dataproc.py

```diff
@@ -21,9 +23,49 @@
 PYFILES_ZIP = 'pyfiles.zip'
 
 
-def hail_dataproc_job(
+class DataprocCluster:
+    """
+    Helper class that represents a Dataproc cluster created within a Batch
+    """
+
+    def __init__(
+        self,
+        batch: hb.Batch,
+        start_job: hb.batch.job.Job,
+        stop_job: hb.batch.job.Job,
+        cluster_id: str,
+        cluster_title: Optional[str] = None,
+    ):
+        self._batch = batch
+        self._start_job = start_job
+        self._stop_job = stop_job
+        self._cluster_id = cluster_id
+        self.cluster_title = cluster_title
```

Comment:
Should this also be private? And how about `cluster_name` instead of `cluster_title`?

```suggestion
        self._cluster_name = cluster_name
```